### PR TITLE
[velero] Pass config trough tpl and implement reload trigger

### PR DESCRIPTION
**Pull request by jkroepke**
#### Special notes for your reviewer:

This PR implements to things:

Implement an optional pod trigger, if the secret has been changed. (Fixes https://github.com/vmware-tanzu/velero/issues/5866)
Implement an flag to pass credentials to helm tpl function.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the values.yaml or README.md
- [x] Title of the PR starts with chart name (e.g. `[velero]`)

### CODE REVIEW

**Review by jenting — COMMENTED** [review]
Does this requires reloader to make the pod automatically restart when the secrets change?

File: charts/velero/values.yaml [hunk 1]
```diff
@@ -497,6 +497,12 @@ credentials:
   # that will be used to load environment variables into velero and node-agent.
   # Secret should be in format - https://kubernetes.io/docs/concepts/configuration/secret/#use-case-as-container-environment-variables
   extraSecretRef: ""
+  # If true, the velero controller will be restarted if the secret has been changed.
+  restartOnChange: false
+  # Pass the secretContents and extraSecretRef configuration directives through Helm's templating engine.
+  # If the configuration contains {{, they'll need to be properly escaped so that they are not interpreted by Helm
+  # ref: https://helm.sh/docs/developing_charts/#using-the-tpl-function
+  tplConfig: false
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply charts/velero/values.yaml]: I remove this. 👍

File: charts/velero/values.yaml [hunk 2]
```diff
@@ -497,6 +497,8 @@ credentials:
   # that will be used to load environment variables into velero and node-agent.
   # Secret should be in format - https://kubernetes.io/docs/concepts/configuration/secret/#use-case-as-container-environment-variables
   extraSecretRef: ""
+  # If true, the velero controller will be restarted if the secret has been changed.
+  restartOnChange: false
```
Comment (jenting) [inline charts/velero/values.yaml:501]: I'd prefer make it as default and remove this configuration in values.yaml.

**Comment by jkroepke** [comment]
> > @jenting No. The technic that I used here is described well at helm docs:
> > [helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments](https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments)
> 
> I think it still depends on the executing the [helm upgrade](https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments:~:text=Depending%20on%20the%20application%20a%20restart%20may%20be%20required%20should%20those%20be%20updated%20with%20a%20subsequent%20helm%20upgrade) so the Pod will automatically restart when the secret change. Otherwise, we usually depends on the [reloader](https://github.com/stakater/Reloader).

It's also GitOps compatible, because FluxCD uses helm upgrade and ArgoCD uses helm template which includes the checksums and forces a reloaded based on the checksums.

It also works, if helm is embedded into kustomize. Grafana Helm charts uses the mechanic, too.

**Comment by jkroepke** [comment]
Hi @jenting 

I changed the checksum logic a bit that only the secrets are included for the checksum generation. This avoid pod-trigger, if the labels has been changed.

**Review by jenting — COMMENTED** [review]
Is it possible to support the case for the 'existingsecret'?
Because the secret might be generated externally, such as Vault. And we should support automatically update the secret sha256 checksum for the existingsecret as well.

**Review by jenting — APPROVED** [review]
Code LGTM

**Comment by jenting** [comment]
@jkroepke could you help rebase this PR and bump the chart version? Thanks.

**Comment by jkroepke** [comment]
@jenting I do a merge from main here, since you do commits here on branch and with rebase, the signature gets invalid.

**Review by jenting — APPROVED** [review]

**Review by qiuming-best — APPROVED** [review]

**Pull request merged**